unumpy/_multimethods.py

```diff
@@ -391,16 +419,109 @@ def ones(shape, dtype=float, order="C"):
     return (mark_dtype(dtype),)
 
 
+@create_numpy(
+    _self_dtype_argreplacer,
+    default=lambda a, dtype=None, order="K", subok=True, shape=None: full_like(
+        a, 1, dtype, order, subok, shape
+    ),
+)
+@all_of_type(ndarray)
+def ones_like(a, dtype=None, order="K", subok=True, shape=None):
+    return (a, mark_dtype(dtype))
+
+
 @create_numpy(_dtype_argreplacer)
 def eye(N, M=None, k=0, dtype=float, order="C"):
     return (mark_dtype(dtype),)
 
 
+@create_numpy(_dtype_argreplacer, default=lambda n, dtype=None: eye(n, dtype=dtype))
+def identity(n, dtype=None):
+    return (mark_dtype(dtype),)
+
+
 @create_numpy(_dtype_argreplacer)
 def asarray(a, dtype=None, order=None):
     return (mark_dtype(dtype),)
 
 
+@create_numpy(_self_dtype_argreplacer)
+@all_of_type(ndarray)
+def asanyarray(a, dtype=None, order=None):
+    return (a, mark_dtype(dtype))
+
+
+@create_numpy(
+    _dtype_argreplacer,
+    default=lambda a, dtype=None: asarray(a, dtype=dtype, order="C"),
+)
+@all_of_type(ndarray)
+def ascontiguousarray(a, dtype=None):
+    return (mark_dtype(dtype),)
+
+
+@create_numpy(_self_argreplacer)
+@all_of_type(ndarray)
+def copy(a, order="K"):
+    return (a,)
+
+
+@create_numpy(_dtype_argreplacer)
+def frombuffer(buffer, dtype=float, count=-1, offset=0):
+    return (mark_dtype(dtype),)
+
+
+@create_numpy(_dtype_argreplacer)
+def fromfile(file, dtype=float, count=-1, sep="", offset=0):
+    return (mark_dtype(dtype),)
+
+
+@create_numpy(_dtype_argreplacer)
+def fromfunction(function, shape, **kwargs):
+    if "dtype" in kwargs:
+        dtype = kwargs["dtype"]
+    else:
+        dtype = float
+
+    return (mark_dtype(dtype),)
+
+
+def _fromiter_default(iterable, dtype, count=-1):
+    if not isinstance(iterable, (list, tuple, range)):
+        iterable = list(iterable)
+    if count == -1:
+        return array(iterable, dtype=dtype)
+    else:
+        return array(iterable[:count], dtype=dtype)
```

Comment:
This won't work for infinite sequences, for example  `itertools.repeat(5)`. Perhaps guard the start with:
```python
if count >= 0:
	iterable = itertools.slice(iterable, 0, count)
```
to make the sequence finite.

Reply:
I have to change the other guard as well then, since doing `list(iterable)` on an infinite sequence makes Python enter an endless loop. Also, the method isn't suppose so accept non-iterable types.

Reply:
I have to change the other guard as well then, since doing list(iterable) on an infinite sequence makes Python enter an endless loop. Also, the method isn't suppose so accept non-iterable types.